tests/integration/test_charm.py

```diff
@@ -0,0 +1,191 @@
+    )
+
+    # Wait for all the charms to become Active.
+    for name in APP_LIST:
+        await ops_test.model.wait_for_idle(apps=[name], status="active", timeout=2000)
+
+    # effectively disable the update status from firing
+    await ops_test.model.set_config({"update-status-hook-interval": "60m"})
+
+
+@pytest.mark.abort_on_fail
+async def test_applications_are_up(ops_test: pytest_plugin.OpsTest):
+    """Test the direct FINOS Legend connection.
+
+    The FINOS Legend application services should be accessible through their IP, ports,
+    and routes.
+    """
+    status = await ops_test.model.get_status()
+
+    for app_name in LEGEND_APPS:
+        unit_name = "%s/0" % app_name
+        address = status["applications"][app_name]["units"][unit_name]["address"]
+
+        url = "http://%s:%s%s" % (address, APP_PORTS[app_name], APP_ROUTES[app_name])
+        can_connect = check_legend_connection(app_name, url)
+
+        assert can_connect, "Could not reach %s through its IP." % app_name
+        logger.info("Successfully reached %s through its IP.", app_name)
+
+
+@pytest.mark.abort_on_fail
+async def test_nginx_ingress_integration(ops_test: pytest_plugin.OpsTest):
+    """Test the FINOS Legend connection through NGINX Ingress.
+
+    The Legend applications have been configured with the legend-host external-hostname by default,
+    and they should be accessible through it due to the nginx-ingress-integrator charm.
+    """
+    # We should now be able to connect to FINOS Legend through it's service-hostname (this bundle
+    # sets it to "legend-host"). In this test scenario, we don't have a resolver for it. One could
+    # configure the /etc/hosts file to have the line:
+    # 127.0.0.1 legend-host
+    # Having the line above would resolve the hostname. For the current testing purposes, we
+    # can simply connect to 127.0.0.1 and having the hostname as a header. This is the
+    # equivalent of:
+    # curl --header 'Host: legend-host' http://127.0.0.1
+
+    await ops_test.model.deploy(NGINX_INGRESS_CHARM, channel="edge", trust=True)
+
+    await ops_test.model.add_relation(NGINX_INGRESS_CHARM, SDLC_APP_NAME)
+    await ops_test.model.add_relation(NGINX_INGRESS_CHARM, ENGINE_APP_NAME)
+    await ops_test.model.add_relation(NGINX_INGRESS_CHARM, STUDIO_APP_NAME)
+
+    for app_name in LEGEND_APPS_CONFIG:
+        url = "http://127.0.0.1%s" % APP_ROUTES[app_name]
+        headers = {"Host": app_name}
+        can_connect = check_legend_connection(app_name, url, headers)
+
+        assert can_connect, "Could not reach %s through its service-hostname." % app_name
+
+        logger.info("Successfully reached %s through its service-hostname.", app_name)
+
+    # configure and check connection to external host
+    sdlc = ops_test.model.applications[SDLC_APP_NAME]
+    await sdlc.set_config({"external-hostname": LEGEND_HOST, "enable-tls": "true"})
+
+    app_name = SDLC_APP_NAME
```

Comment:
That doesn't look right. You're changing the ``external-hostname`` above, which means that the SDLC application will no longer be reachable through it's previous hostname (which by default is the application name), and it should be reachable through the new hostname instead, which you've set to ``legend-host``.

Reply:
Solved.